record/record.go

```diff
@@ -144,23 +146,29 @@ func (r *Reader) nextChunk(wantFirst bool) error {
 			chunkType := r.buf[r.j+6]
 
 			if checksum == 0 && length == 0 && chunkType == 0 {
-				if wantFirst {
+				if wantFirst || r.recovering {
 					// Skip the rest of the block, if it looks like it is all zeroes.
 					// This is common if the record file was created via mmap.
-					r.i = r.n
-					r.j = r.n
+					r.Recover()
```

Comment:
So WDYT of my solution. Before calling Recover, i set r.err = errors.New("leveldb/record: block appears to be zeroed")

It's short lived, but it self documents and addresses the problem you pointed out.